crates/miden-testing/tests/auth/multisig.rs

```diff
@@ -0,0 +1,414 @@
+        mock_chain.add_pending_executed_transaction(&executed_tx)?;
+        mock_chain.prove_next_block()?;
+    }
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_multisig_replay_protection() -> anyhow::Result<()> {
+    // Test 2/3 multisig where tx is executed, then attempted again (should fail on 2nd attempt)
+
+    // Setup keys and authenticators (3 approvers, but only 2 signers)
+    let (_secret_keys, public_keys, authenticators) = setup_keys_and_authenticators(3, 2)?;
+
+    // Create 2/3 multisig account
+    let multisig_account = create_multisig_account(2, &public_keys, 20)?;
+
+    let mut mock_chain = MockChainBuilder::with_accounts([multisig_account.clone()])
+        .unwrap()
+        .build()
+        .unwrap();
+
+    mock_chain.prove_next_block()?;
+
+    let salt = Word::from([Felt::new(3); 4]);
+
+    // Execute transaction without signatures first to get tx summary
+    let tx_context_init = mock_chain
+        .build_tx_context(multisig_account.id(), &[], &[])?
+        .auth_args(salt)
+        .build()?;
+
+    let tx_summary = match tx_context_init.execute().await.unwrap_err() {
+        TransactionExecutorError::Unauthorized(tx_effects) => tx_effects,
+        error => panic!("expected abort with tx effects: {error:?}"),
+    };
+
+    // Get signatures from 2 of the 3 approvers
+    let msg = tx_summary.as_ref().to_commitment();
+    let tx_summary = SigningInputs::TransactionSummary(tx_summary);
+
+    let sig_1 = authenticators[0].get_signature(public_keys[0].into(), &tx_summary).await?;
+    let sig_2 = authenticators[1].get_signature(public_keys[1].into(), &tx_summary).await?;
+
+    // Populate advice map with signatures
+    let mut advice_map = AdviceMap::default();
+    advice_map.insert(Hasher::merge(&[public_keys[0].into(), msg]), sig_1);
+    advice_map.insert(Hasher::merge(&[public_keys[1].into(), msg]), sig_2);
+
+    // Execute transaction with signatures - should succeed (first execution)
+    let tx_context_execute = mock_chain
+        .build_tx_context(multisig_account.id(), &[], &[])?
+        .extend_advice_map(advice_map.iter().map(|(k, v)| (*k, v.to_vec())))
+        .auth_args(salt)
+        .build()?;
+
+    let executed_tx = tx_context_execute.execute().await.expect("First transaction should succeed");
+
+    // Apply the transaction to the mock chain
+    mock_chain.add_pending_executed_transaction(&executed_tx)?;
+    mock_chain.prove_next_block()?;
+
+    // Now attempt to execute the same transaction again - should fail due to replay protection
+    let tx_context_replay = mock_chain
+        .build_tx_context(multisig_account.id(), &[], &[])?
+        .extend_advice_map(advice_map.iter().map(|(k, v)| (*k, v.to_vec())))
+        .auth_args(salt)
+        .build()?;
+
+    // This should fail - due to replay protection
+    let result = tx_context_replay.execute().await;
+    assert!(result.is_err(), "Second execution of the same transaction should fail");
```

Comment:
```suggestion
    let result = tx_context_replay.execute().await;
    assert_transaction_executor_error!(result, ERR_TX_ALREADY_EXECUTED);
```

If you change the name of the error constant, then we can import it and assert the exact error here.